src/commands/DensityProfile.cpp

```diff
@@ -0,0 +1,191 @@
+// cfiles, an analysis frontend for the Chemfiles library
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include <docopt/docopt.h>
+#include <sstream>
+#include <fstream>
+#include <iostream>
+
+#include "DensityProfile.hpp"
+#include "Errors.hpp"
+#include "utils.hpp"
+#include "geometry.hpp"
+
+using namespace chemfiles;
+
+static const double pi = 3.141592653589793238463;
+static const char OPTIONS[] =
+R"(Compute the density profile of particles along a given axis or radially. 
+The output for the radial density profile is normalized by r.
+Selections for the particles can be specified using the chemfiles selection 
+language. It is possible to provide an alternative unit cell or topology 
+for the trajectory file if they are not defined in the trajectory format.
+The axis can be specified using a coordinate vector (e.g. z axis would be (0,0,1)).
+
+For more information about chemfiles selection language, please see
+http://chemfiles.org/chemfiles/latest/selections.html
+
+Usage:
+  cfiles density [options] <trajectory>
+  cfiles density (-h | --help)
+
+Examples:
+  cfiles density water.xyz --cell 15:15:25 --guess-bonds --axis=1,1,1
+  cfiles density in.pdb --selection="x>3"
+
+Options:
+  -h --help                     show this help
+  -o <file>, --output=<file>    write result to <file>. This default to the
+                                trajectory file name with the `.density.dat` extension.
+  --format=<format>             force the input file format to be <format>
+  -t <path>, --topology=<path>  alternative topology file for the input
+  --topology-format=<format>    use <format> as format for the topology file
+  --guess-bonds                 guess the bonds in the input
+  -c <cell>, --cell=<cell>      alternative unit cell. <cell> format is one of
+                                <a:b:c:α:β:γ> or <a:b:c> or <a>. 'a', 'b' and
+                                'c' are in angstroms, 'α', 'β', and 'γ' are in
+                                degrees.
+  --steps=<steps>               steps to use from the input. <steps> format
+                                is <start>:<end>[:<stride>] with <start>, <end>
+                                and <stride> optional. The used steps goes from
+                                <start> to <end> (excluded) by steps of
+                                <stride>. The default values are 0 for <start>,
+                                the number of steps for <end> and 1 for 
+                                <stride>.
+  -s <sel>, --selection=<sel>   selection to use for the particles. This must be a
+                                selection of size 1.
+                                [default: atoms: all]
+  --profile=<str>               type of density profile. Currently implemented are
```

Comment:
What do you think of having a flag to go from linear to radial? `cfiles density --axis X` would be linear, and `cfiles density --axis X --radial` would be radial.